x/mongo/driver/session/client_session.go

```diff
@@ -95,6 +95,7 @@ type LoadBalancedTransactionConnection interface {
 	Description() description.Server
 	Close() error
 	ID() string
+	ServerConnectionID() *uint32
```

Comment:
Because `LoadBalancedTransactionConnection` implements the `PinnedConnection` interface which contains a `Connection` interface, I had to add `ServerConnectionID` to `LoadBalancedTransactionConnection`. I had to do the same thing for various testing types (`mockConnection`, `ChannelConn` and `mtest.connection`).